x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -195,8 +199,11 @@ export const getDefaultTab = (
   const kspmTotalFindings = kspmStats?.stats.totalFindings;
   const installedPolicyTemplatesCspm = pluginStatus?.cspm?.status;
   const installedPolicyTemplatesKspm = pluginStatus?.kspm?.status;
-  let preferredDashboard = POSTURE_TYPE_CSPM;
+  let preferredDashboard: PosturePolicyTemplate = POSTURE_TYPE_CSPM;
 
+  if (!cspmStats || !pluginStatus) {
+    return;
+  }
```

Comment:
shouldn't kspmStats be checked as well? 
what happens of the one of those calls failed? do we get an empty screen or the error log?

Reply:
When first routing to Dashboard page, CSPM stats and KSPM stats are undefined. There is no need to check if kspmStats are defined. This check helps resolves race condition when we are waiting for cspm stats and kspm stats to be defined then show  the correct tabs based on plugin status, cspm and kspm status.